### PR TITLE
[ADD]Hr material:added new module hr_material,which allows you to easily...

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
... keep track of office materials.
- Task: https://www.odoo.com/web#id=6897&view_type=form&model=project.task

### CODE REVIEW

File: addons/hr/res_config.py [hunk 1]
```diff
@@ -48,8 +48,8 @@ class hr_config_settings(osv.osv_memory):
             help ="""This installs the module account_analytic_analysis, which will install sales management too."""),
         'module_hr_payroll': fields.boolean('Manage payroll',
             help ="""This installs the module hr_payroll."""),
-        'module_website_hr_recruitment': fields.boolean('Publish jobs on your website',
-            help ="""This installs the module website_hr_recruitment"""),
```
Comment (rim-odoo) [inline addons/hr/res_config.py]: Why do you remove the ability to publish jobs on the website? It has nothing to do with tracking materials...

File: addons/hr/res_config_view.xml [hunk 2]
```diff
@@ -99,12 +99,13 @@
...
                     </group>
-                    <group name="website_grp">
-                        <label for="id" string="Website"/>
+
+                    <group name="hr_materials_grp">
+                        <label for="id" string="Materials Tracking"/>
                         <div>
-                            <div>
-                                <field name="module_website_hr_recruitment" class="oe_inline"/>
-                                <label for="module_website_hr_recruitment"/>
```
Comment (rim-odoo) [inline addons/hr/res_config_view.xml]: same question here, why removing?

File: addons/hr_material/static/src/js/hr_material.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+openerp.hr_material = function(openerp) {
+    openerp.web_kanban.KanbanRecord.include({
```
Comment (rim-odoo) [inline addons/hr_material/static/src/js/hr_material.js]: do not forget `use strict`;

File: addons/hr_material/views/hr_material.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (rim-odoo) [inline addons/hr_material/views/hr_material.xml]: with <odoo> you do not need <data> anymore I think

File: addons/hr_material/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    'data': [
+        'security/hr_security.xml',
+        'security/ir.model.access.csv',
+        'security/ir_rule.xml',
+        'data/hr_material_data.xml',
+        'views/res_config_view.xml',
+        'views/hr_material_view.xml',
+        'views/hr_material.xml',
+    ],
+    'auto_install': False,
```
Comment (rim-odoo) [inline addons/hr_material/__openerp__.py]: False is the default, you can remove this key

File: addons/hr_material/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    'installable': True,
+    'application': True,
+}
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (rim-odoo) [inline addons/hr_material/__openerp__.py]: No more modelines please! Never!

File: addons/hr/res_config.py [hunk 7]
```diff
@@ -48,8 +48,8 @@ class hr_config_settings(osv.osv_memory):
...
+        'module_hr_material': fields.boolean('Manage several material categories like IT, Office material, Engine, ...',
```
Comment (tde-banana-odoo) [inline addons/hr/res_config.py]: We do not propose to install modules in setting anymore; setting are there to configure things or choose options. So please remove this.

File: addons/hr_material/__openerp__.py (hunk 5 above)
Comment (tde-banana-odoo) [inline addons/hr_material/__openerp__.py]: Up.

File: addons/hr_material/__openerp__.py (hunk 6 above)
Comment (tde-banana-odoo) [inline addons/hr_material/__openerp__.py]: Up.

File: addons/hr/res_config_view.xml [hunk 8]
```diff
@@ -99,12 +99,13 @@
...
+                            <div name="hr_materials">
```
Comment (tde-banana-odoo) [inline addons/hr/res_config_view.xml]: Remove this with the field.

File: addons/hr_material/data/hr_material_data.xml [hunk 9]
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+
+        <!-- Standard stages for Maintenance Request -->
+        <record id="stage_new" model="hr.material.stage">
```
Comment (tde-banana-odoo) [inline addons/hr_material/data/hr_material_data.xml]: ids should be more neutral, rename them to stage_0, stage_1, ...

File: addons/hr_material/data/hr_material_data.xml [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+            <field name="fold" eval="True" />
+        </record>
+        <record id="stage_cancel" model="hr.material.stage">
+            <field name="name">Cancelled</field>
+            <field name="sequence" eval="10" />
+            <field name="fold" eval="True" />
+        </record>
+
+        <record id="mail_alias_material" model="mail.alias">
+            <field name="alias_name">internal.material</field>
```
Comment (tde-banana-odoo) [inline addons/hr_material/data/hr_material_data.xml]: Defining a default alias is risky. If people do not pay attention they will have emails going to this address that is not necessarily configured in their email servers. Is it really necessary ? It should be blank by default IMHO.

File: addons/hr_material/data/hr_material_data.xml [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+            <field name="tag">reload</field>
+            <field name="params" eval="{'menu_id': ref('hr.menu_hr_root')}"/>
+        </record>
+        <record id="base.open_menu" model="ir.actions.todo">
+            <field name="action_id" ref="action_client_hr_menu"/>
+            <field name="state">open</field>
+        </record>
+
+        <!--Material categories-->
+        <record id="hr_material_computer" model="hr.material.category">
```
Comment (tde-banana-odoo) [inline addons/hr_material/data/hr_material_data.xml]: Why having data that look like demo data to me ?

File: addons/hr_material/models/hr_material.py [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+            'hr_material.mt_mat_assign': lambda self, cr, uid, obj, ctx = None: obj.department_id,
+        },
+    }
+
+    @api.one
+    def _maintenance_count(self):
+        self.maintenance_count = len(self.maintenance_ids)
+
+    @api.depends('name', 'serial_no')
+    def _get_full_name(self):
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: Should be a name_get, with a name_search probably.

File: addons/hr_material/models/hr_material.py [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+
+class HrMaterialCategory(models.Model):
+    _name = 'hr.material.category'
+    _inherits = {"mail.alias": "alias_id"}
+    _inherit = ['mail.thread']
+    _description = 'Material Category'
+
+    @api.one
+    def _compute_material_count(self):
+        self.material_count = len(self.material_ids)
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: multi + read_group maybe ?

File: addons/hr_material/models/hr_material.py [hunk 14]
```diff
@@ -0,0 +1,342 @@
...
+        self.fold = False if self.material_count else True
+
+    @api.one
+    def _compute_maintenance_count(self):
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: multi + read_group maybe ?

File: addons/hr_material/models/hr_material.py [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+        self.maintenance_count = len(self.maintenance_ids)
+
+    name = fields.Char('Category Name', required=True, translate=True)
+    user_id = fields.Many2one('res.users', 'Responsible', track_visibility='onchange', default=lambda self: self.env.uid)
+    color = fields.Integer('Color Index')
+    note = fields.Text('Comments', translate=True)
+    material_ids = fields.One2many('hr.material', 'category_id', string='Materials', copy=False)
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: This field does not seem necessary.

File: addons/hr_material/models/hr_material.py [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+    material_count = fields.Integer(compute='_compute_material_count', string="Material")
+    maintenance_ids = fields.One2many('hr.material.request', 'category_id', copy=False, domain=[('stage_id.fold', '=', False)])
+    maintenance_count = fields.Integer(compute='_compute_maintenance_count', string="Maintenance")
+    alias_id = fields.Many2one(
+        'mail.alias', 'Alias', ondelete="restrict", required=True,
+        help="Email alias for this material category. New emails will automatically "
+        "create new maintenance request for this material category.")
+    fold = fields.Boolean(compute='_compute_material_count', string='Folded in Kanban View')
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: Make its own method + stored + depends.

File: addons/hr_material/models/hr_material.py [hunk 17]
```diff
@@ -0,0 +1,342 @@
...
+        category_id.alias_id.write({'alias_parent_thread_id': category_id.id, 'alias_defaults': {'category_id': category_id.id}})
+        return category_id
+
+    @api.multi
+    def unlink(self):
+        alias_ids = []
+        for category in self:
+            if category.material_ids or category.maintenance_ids:
+                raise Warning(_('You cannot delete a Material Category containing Material/Maintenance Request. You can either delete all the material/maintenance which belongs to in this category and then delete the material category.'))
+            elif category.alias_id:
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: What's the purpose of this condition ? aliases should be deleted with an ondelete=cascade, nothing to check.

File: addons/hr_material/models/hr_material.py [hunk 18]
```diff
@@ -0,0 +1,342 @@
...
+    name = fields.Char('Category Name', required=True, translate=True)
+    user_id = fields.Many2one('res.users', 'Responsible', track_visibility='onchange', default=lambda self: self.env.uid)
+    color = fields.Integer('Color Index')
+    note = fields.Text('Comments', translate=True)
+    material_ids = fields.One2many('hr.material', 'category_id', string='Materials', copy=False)
+    material_count = fields.Integer(compute='_compute_material_count', string="Material")
+    maintenance_ids = fields.One2many('hr.material.request', 'category_id', copy=False, domain=[('stage_id.fold', '=', False)])
+    maintenance_count = fields.Integer(compute='_compute_maintenance_count', string="Maintenance")
+    alias_id = fields.Many2one(
+        'mail.alias', 'Alias', ondelete="restrict", required=True,
```
Comment (tde-banana-odoo) [inline addons/hr_material/models/hr_material.py]: Set to cascade.

File: addons/hr/res_config.py (hunk 1 above)
Comment (avo-odoo) [inline addons/hr/res_config.py]: I have added back this field.

File: addons/hr/res_config_view.xml (hunk 2 above)
Comment (avo-odoo) [inline addons/hr/res_config_view.xml]: Done !.

File: addons/hr_material/static/src/js/hr_material.js (hunk 3 above)
Comment (avo-odoo) [inline addons/hr_material/static/src/js/hr_material.js]: Done.

File: addons/hr_material/views/hr_material.xml (hunk 4 above)
Comment (avo-odoo) [inline addons/hr_material/views/hr_material.xml]: Done

File: addons/hr_material/__openerp__.py (hunk 5 above)
Comment (avo-odoo) [inline addons/hr_material/__openerp__.py]: Done.

File: addons/hr_material/__openerp__.py (hunk 6 above)
Comment (avo-odoo) [inline addons/hr_material/__openerp__.py]: Done.

File: addons/hr/res_config.py (hunk 7 above)
Comment (avo-odoo) [inline addons/hr/res_config.py]: Done

File: addons/hr/res_config_view.xml (hunk 8 above)
Comment (avo-odoo) [inline addons/hr/res_config_view.xml]: Done

File: addons/hr_material/data/hr_material_data.xml (hunk 9 above)
Comment (avo-odoo) [inline addons/hr_material/data/hr_material_data.xml]: Done.

File: addons/hr_material/data/hr_material_data.xml (hunk 11 above)
Comment (avo-odoo) [inline addons/hr_material/data/hr_material_data.xml]: I have moved this data in demo file.

File: addons/hr_material/models/hr_material.py (hunk 12 above)
Comment (avo-odoo) [inline addons/hr_material/models/hr_material.py]: Done

File: addons/hr_material/models/hr_material.py (hunk 13 above)
Comment (avo-odoo) [inline addons/hr_material/models/hr_material.py]: Done.

File: addons/hr_material/models/hr_material.py (hunk 14 above)
Comment (avo-odoo) [inline addons/hr_material/models/hr_material.py]: Done

File: addons/hr_material/models/hr_material.py (hunk 16 above)
Comment (avo-odoo) [inline addons/hr_material/models/hr_material.py]: Done.!

File: addons/hr_material/models/hr_material.py (hunk 17 above)
Comment (avo-odoo) [inline addons/hr_material/models/hr_material.py]: Done.

File: addons/hr_material/models/hr_material.py (hunk 18 above)
Comment (avo-odoo) [inline addons/hr_material/models/hr_material.py]: Done.

File: addons/hr_material/data/hr_material_data.xml (hunk 10 above)
Comment (avo-odoo) [inline addons/hr_material/data/hr_material_data.xml]: This alias we used in HrMaterialRequest kanban view help default when none of record in kanban view.so suggest me i have removed this mail.alias or not ?

**Comment by rim-odoo** [comment]
Due to bad branch manipulation, work continues in #5712

**Pull request closed without merging**